### PR TITLE
Adopt Black for Python.

### PR DESCRIPTION
**Pull request by thorstenhater**
Add black formatter

### CODE REVIEW

**Comment by brenthuisman** [comment]
Building docs is not part of our tests, but relies on much of our Python code. Can you confirm it builds fine?

**Comment by thorstenhater** [comment]
They do. All the literal include lines are in jeopardy though.

**Comment by Helveg** [comment]
What about an [`--ignore-revs-file`](https://black.readthedocs.io/en/stable/guides/introducing_black_to_your_project.html)?

**Comment by thorstenhater** [comment]
We can do that, but -- I assume -- not before merging this, since we need to commit IDs?

**Comment by thorstenhater** [comment]
Then I can only say thank you for the hint, will do ;)

**Review by Helveg — APPROVED** [review]
I agree with black 👍 This makes the `'"label"'` syntax slightly weirder

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: "Lint"
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
```
Comment (brenthuisman) [inline .github/workflows/lint.yml:13]: Maybe instead use their Action: https://black.readthedocs.io/en/stable/integrations/github_actions.html

Also, maybe make this a monthly job and have this Action check on each PR: https://github.com/marketplace/actions/python-black-formatter-only-on-edited-files
It only check changed files, so it won't put people on the hook for code unrelated to what they are changing.

File: doc/contrib/coding-style.rst [hunk 2]
```diff
@@ -3,6 +3,26 @@
...
+
+.. code::
+
+   # Install the formatter if not present
+   pip install black
+   # Automatically apply style. If unsure what this does read on.
+   black --extend-exclude '/(ext|python/pybind11|doc/scripts/.*_theme)' . scripts/build-catalogue.in
+
+The formatter can also be run with ``--check`` to list offending files and
+``--diff`` to preview changes.
```
Comment (brenthuisman) [inline doc/contrib/coding-style.rst:21]: Let's mention how people can set their editor to do this for them: https://black.readthedocs.io/en/stable/integrations/editors.html

File: python/__init__.py [hunk 3]
```diff
@@ -9,12 +9,14 @@
...
+
     here = os.path.abspath(os.path.dirname(__file__))
-    with open(os.path.join(here, 'VERSION')) as version_file:
+    with open(os.path.join(here, "VERSION")) as version_file:
         return version_file.read().strip()
 
+
 __version__ = get_version()
-__config__  = config()
+__config__ = config()
```
Comment (brenthuisman) [inline python/__init__.py:19]: No more manual vertical aligning, I hope you can actually live with that ;)

**Review by brenthuisman — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (brenthuisman) [reply .github/workflows/lint.yml:13]: Even better/worse: have a bot autopush blacked files to a PR: https://github.com/psf/black/blob/2234639a31ed8aeca05e20aa6966005277e9ae07/.github/workflows/autoblack.yml

Comment (thorstenhater) [reply .github/workflows/lint.yml:13]: Using their action is great though.

Comment (brenthuisman) [reply .github/workflows/lint.yml:13]: You're right: by merging this PR there should be no way black will find errors in other areas but those touched in future PRs.

Having the bot autocorrect is neat though, and scary. What do you think?

File: doc/contrib/coding-style.rst (hunk 2 above)
Comment (thorstenhater) [reply doc/contrib/coding-style.rst]: 👍

**Pull request merged**